generator/templates/client/client.gotmpl

```diff
@@ -25,7 +25,7 @@ import (
 )
 
 // New creates a new {{ humanize .Name }} API client.
-func New(transport runtime.ClientTransport, formats strfmt.Registry) *Client {
+func New(transport runtime.ClientTransport, formats strfmt.Registry) ClientService {
```

Comment:
I've ended up using a more dynamic interface name instead of `ClientService`.

```
{{ pascalize .Name }}Client
```

This way there is less ambiguity when there are multiple clients in the codebase.

Reply:
@rokf I am really not sure this is a good idea: when organizing operations with tags, we end up generating `{{ pascalize .Name }}` packages. IMHO, the best way to avoid ambiguity is to generate client with an appropriate package target, so we use it like `target.ClientService`.